### PR TITLE
support for generic assets

### PR DESCRIPTION
**Pull request by cdoron**
Signed-off-by: Doron Chen <cdoron@il.ibm.com>

### CODE REVIEW

File: pkg/openmetadata-connector-core/connector_service.go [hunk 1]
```diff
@@ -97,21 +95,23 @@ func (s *OpenMetadataAPIService) CreateAsset(ctx context.Context, //nolint
...
+		_, found = s.findIngestionPipeline(ctx, c, ingestionPipelineNameFull)
 
-	if !found {
-		// Let us create an ingestion pipeline
-		s.logger.Info().Msg("Ingestion Pipeline not found. Creating.")
-		_, _ = s.createIngestionPipeline(ctx, c, databaseServiceID, ingestionPipelineName)
+		if !found {
+			// Let us create an ingestion pipeline
+			s.logger.Info().Msg("Ingestion Pipeline not found. Creating.")
+			_, _ = s.createIngestionPipeline(ctx, c, databaseServiceID, ingestionPipelineName)
```
Comment (roytman) [inline pkg/openmetadata-connector-core/connector_service.go:112]: we don't check the results ?

File: pkg/database-types/constants.go [hunk 2]
```diff
@@ -11,6 +11,8 @@ const AwsSecretAccessKey = "awsSecretAccessKey"
 const Bucket = "bucket"
 const BucketName = "bucketName"
 const ConfigSource = "configSource"
+const ConnectionOptions = "connectionOptions"
+const CustomDatabase = "CustomDatabase"
```
[the remark this answers is not in the document]
Comment (cdoron) [reply pkg/database-types/constants.go:15]: That's right.

File: pkg/openmetadata-connector-core/connector_service.go (hunk 1 above)
Comment (cdoron) [reply pkg/openmetadata-connector-core/connector_service.go:112]: That's right. We can move forward even if the ingestion pipeline was not created properly.

File: pkg/database-types/generic.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+// OM databaseService-s of type CustomDatabase expect the configuration key-value pairs to be placed
+// within the 'connectionOptions' field. The values in the key-value pairs must be strings
+func (m *generic) TranslateFybrikConfigToOpenMetadataConfig(config map[string]interface{}, credentials *string) map[string]interface{} {
+	ret := make(map[string]string)
+	for key, value := range config {
+		valueStr, ok := value.(string)
+		if ok {
+			ret[key] = valueStr
+		} else {
+			jsonStr, _ := json.Marshal(value)
```
[the remark this answers is not in the document]
Comment (cdoron) [reply pkg/database-types/generic.go:34]: We could, for instance if someone changes the connection parameters through the OpenMetadata UI.
I am adding a log message when the `json.Marshal` operation fails.

**Review by roytman — APPROVED** [review]

**Pull request merged**